pkg/objstore/objstore.go

```diff
@@ -200,15 +200,21 @@ func DownloadFile(ctx context.Context, logger log.Logger, bkt BucketReader, src,
 }
 
 // DownloadDir downloads all object found in the directory into the local directory.
-func DownloadDir(ctx context.Context, logger log.Logger, bkt BucketReader, src, dst string) error {
+func DownloadDir(ctx context.Context, logger log.Logger, bkt BucketReader, src, dst string, ignoredPaths []string) error {
 	if err := os.MkdirAll(dst, 0777); err != nil {
 		return errors.Wrap(err, "create dir")
 	}
 
 	var downloadedFiles []string
 	if err := bkt.Iter(ctx, src, func(name string) error {
 		if strings.HasSuffix(name, DirDelim) {
-			return DownloadDir(ctx, logger, bkt, name, filepath.Join(dst, filepath.Base(name)))
+			return DownloadDir(ctx, logger, bkt, name, filepath.Join(dst, filepath.Base(name)), ignoredPaths)
+		}
+		for _, ignoredPath := range ignoredPaths {
+			if name == ignoredPath {
+				level.Info(logger).Log("msg", "not downloading again because the hash of the file on disk matches", "file", name)
```

Comment:
This is a generic implementation which just takes in input `ignoredPaths`. In my opinion, this function should know nothing about the fact it will be called with `ignoredPaths` based on content hash comparison. For this reason, I would generalise this message to not mention the hashing thing.